packages/clerk-js/src/v4/UserProfile/MfaPhoneCodePage.tsx

```diff
@@ -0,0 +1,129 @@
+import { FormButtonContainer } from './FormButtons';
+import { NavigateToFlowStartButton } from './NavigateToFlowStartButton';
+import { ContentPage } from './Page';
+import { AddPhone, VerifyPhone } from './PhonePage';
+import { SuccessPage } from './SuccessPage';
+import { AddBlockButton } from './UserProfileBlockButtons';
+
+export const MfaPhoneCodePage = withCardStateProvider(() => {
+  const title = 'Add multifactor authentication';
+
+  const ref = React.useRef<PhoneNumberResource | undefined>();
+  const wizard = useWizard({ defaultStep: 2 });
+
+  return (
+    <Wizard {...wizard.props}>
+      <AddPhone
+        title={title}
+        resourceRef={ref}
+        onSuccess={wizard.nextStep}
+      />
+      <VerifyPhone
+        title={title}
+        resourceRef={ref}
+        onSuccess={wizard.nextStep}
+      />
+      <AddMfa
+        onSuccess={wizard.nextStep}
+        onAddPhoneClick={() => wizard.goToStep(0)}
+        onUnverifiedPhoneClick={phone => {
+          ref.current = phone;
+          wizard.goToStep(1);
+        }}
+        title={title}
+      />
+      <SuccessPage
+        title={title}
+        text={`SMS code multifactor authentication is now enabled for this phone number. When signing in, you will need to enter a verification code sent to this phone number as an additional step.`}
+      />
+    </Wizard>
+  );
+});
+
+type AddMfaProps = {
+  onAddPhoneClick: React.MouseEventHandler;
+  onUnverifiedPhoneClick: (phone: PhoneNumberResource) => void;
+  onSuccess: () => void;
+  title: string;
+};
+
+const AddMfa = (props: AddMfaProps) => {
+  const { onSuccess, title, onAddPhoneClick, onUnverifiedPhoneClick } = props;
+  const card = useCardState();
+  const user = useCoreUser();
+  const availableMethods = user.phoneNumbers.filter(p => !p.reservedForSecondFactor);
+
+  const enableMfa = (phone: PhoneNumberResource) => {
+    if (phone.verification.status !== 'verified') {
+      return onUnverifiedPhoneClick(phone);
+    }
+
+    card.setLoading(phone.id);
+    phone
+      .setReservedForSecondFactor({ reserved: true })
+      .then(() => {
+        card.setIdle();
+        onSuccess();
+      })
+      .catch(err => handleError(err, [], card.setError));
+  };
+
+  return (
+    <ContentPage.Root headerTitle={title}>
+      <Text>
+        {availableMethods.length
+          ? 'Select a phone number to register for SMS code two-step authentication.'
+          : 'There are no available phone numbers to register for SMS code two-step authentication.'}
+      </Text>
+      <Col gap={2}>
+        {availableMethods.map(phone => {
+          const formattedPhone = stringToFormattedPhoneString(phone.phoneNumber);
+          const flag = getFlagEmojiFromCountryIso(parsePhoneString(phone.phoneNumber).iso);
+
+          return (
+            <ArrowBlockButton
+              // elementDescriptor={descriptors.socialButtonsButtonBlock}
```

Comment:
❓ 
Are these comments here by mistake ?

Reply:
This pre-existed, I just moved it to a different file

cc @nikosdouvlis

Reply:
Don't worry about them for not, I've already applies most elementDescriptor on the next branch directly. I 'll do a final pass today to apply the rest now that UP is done